__tests__/mapper_test.re

```diff
@@ -21,26 +21,68 @@ describe("Mappers", () => {
     });
   };
 
+  describe("mmc1", () => {
+    let init = () => make(MMC1, 2, 8);
+
+    let write = (mapper: Mapper.t, address, value) => {
+      for (i in 0 to 4) {
+        mapper.set_prg(address, Util.read_bit(value, i) ? 1 : 0);
+      };
+    };
+
+    test("it can read from chr", () => {
+      let mapper = init();
+
+      expect((mapper.get_chr(0), mapper.get_chr(0x1000)))
+      |> toEqual((0, 0));
+    });
+
+    test("it can switch the lower chr bank", () => {
+      let mapper = init();
+
+      write(mapper, 0xa000, 5);
+
+      expect((mapper.get_chr(0), mapper.get_chr(0x1000)))
+      |> toEqual((5, 0));
+    });
+
+    test("it can switch the upper chr bank", () => {
+      let mapper = init();
+
+      write(mapper, 0xc000, 3);
+
+      expect((mapper.get_chr(0), mapper.get_chr(0x1000)))
+      |> toEqual((0, 3));
+    });
+
+    test("it can switch both chr banks", () => {
+      let mapper = init();
+
+      write(mapper, 0x8000, 0b10000);
+      write(mapper, 0xa000, 2);
+
+      expect((mapper.get_chr(0), mapper.get_chr(0x1000)))
+      |> toEqual((2, 3));
+    });
+  });
+
   describe("nrom", () => {
     let init = () => make(NROM, 1, 1);
 
     test("it can read from prg", () => {
       let mapper = init();
-
-      expect(mapper#get_prg(100)) |> toEqual(0);
+      expect(mapper.get_prg(100)) |> toEqual(0);
```

Comment:
I got _real_ tired of autofmt condensing all of the whitespace out of object implementations, and found it was making MMC1 almost unreadable, so I swapped them over to records. This _seems_ easier to work with to me, but I would like to circle back and compare and profile the generated JS code at some point.